client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -110,35 +143,32 @@
                  :on-success [:scenarios/load-scenario])}))
 
 
-(rf/reg-event-db
+(rf/reg-event-fx
  :scenarios/clear-current-scenario
  in-scenarios
- (fn [db [_]]
-   (assoc db
-          :current-scenario nil
-          :view-state :current-scenario)))
+ (fn [{:keys [db]} [_]]
+   {:db       (assoc db :current-scenario nil)
+    :dispatch [:scenarios/reset-state]}))
 
 ;; Editing scenario
 
-(rf/reg-event-db
+(rf/reg-event-fx
  :scenarios/open-rename-dialog
  in-scenarios
- (fn [db [_]]
+ (fn [{:keys [db]} [_]]
    (let [name (get-in db [:current-scenario :name])]
-     (assoc db
-            :view-state :rename-dialog
-            :rename-dialog {:value name}))))
+     {:db       (assoc db :rename-dialog {:value name})
+      :dispatch-n [[:scenarios/pop-view]
+                   [:scenarios/push-view :rename-dialog]]})))
 
-(rf/reg-event-db
+(rf/reg-event-fx
  :scenarios/cancel-dialog
  in-scenarios
- (fn [db [_]]
-   (let [cancel-next-state (get-in db [:cancel-next-state])]
-     (assoc db
-            :view-state        (or cancel-next-state :current-scenario)
-            :cancel-next-state nil
-            :changeset-dialog  nil
-            :rename-dialog     nil))))
+ (fn [{:keys [db]} [_]]
+   {:db (assoc db
+               :changeset-dialog nil
+               :rename-dialog nil)
```

Comment:
These two look like they belong with the view state. I think it would be safe to move the value resets to the `reset-view-state` function.